src/script/lua_api/l_metadata.cpp

```diff
@@ -51,6 +51,46 @@ MetaDataRef* MetaDataRef::checkobject(lua_State *L, int narg)
 
 // Exported functions
 
+// contains(self, name)
+int MetaDataRef::l_contains(lua_State *L)
+{
+	MAP_LOCK_REQUIRED;
+
+	MetaDataRef *ref = checkobject(L, 1);
+	std::string name = luaL_checkstring(L, 2);
+
+	Metadata *meta = ref->getmeta(false);
+	if (meta == NULL) {
+		lua_pushlstring(L, "", 0);
+		return 1;
+	}
+
+	lua_pushboolean(L, meta->contains(name));
+	return 1;
+}
+
+// get_string_or_nil(self, name)
+int MetaDataRef::l_get_string_or_nil(lua_State *L)
+{
+	MAP_LOCK_REQUIRED;
+
+	MetaDataRef *ref = checkobject(L, 1);
+	std::string name = luaL_checkstring(L, 2);
+
+	Metadata *meta = ref->getmeta(false);
+	if (meta == NULL) {
+		lua_pushlstring(L, "", 0);
```

Comment:
:thinking:

shouldn't this just throw an error or something?

Reply:
return 0 ?

Reply:
@nerzhul That's the same, but yes.